src/mmw/js/src/draw/views.js

```diff
@@ -188,6 +200,66 @@ var ResetDrawView = Marionette.ItemView.extend({
     }
 });
 
+var StreamSliderView = Marionette.ItemView.extend({
+    template: streamSliderTmpl,
+
+    ui: {
+        slider: '#stream-slider',
+        displayValue: '#stream-value'
+    },
+
+    events: {
+        'input @ui.slider': 'onSliderDragged',
+        'change @ui.slider': 'onSliderChanged'
+    },
+
+    initialize: function() {
+        this.streamLayers = settings.getSettings().stream_layers;
+    },
+
+    onShow: function() {
+        this.onSliderDragged();
+    },
+
+    getSliderIndex: function() {
+        return parseInt(this.ui.slider.val());
+    },
+
+    getSliderDisplay: function() {
+        var ind = this.getSliderIndex();
+        if (ind == 0) {
+            return 'Off';
+        } else {
+            return this.streamLayers[ind-1].display;
+        }
+    },
+
+    onSliderDragged: function() {
+        // Preview slider value while dragging.
+        this.ui.displayValue.text(this.getSliderDisplay());
+    },
+
+    onSliderChanged: function() {
+        var ind = this.getSliderIndex();
+        clearStreamLayer(this.model);
+        if (ind > 0) {
+            var streamLayer = this.streamLayers[ind-1];
+            changeStreamLayer(streamLayer.endpoint, streamLayer.table_name, this.model);
```

Comment:
You shouldn't be using the table name anywhere in client side code. We designed the base_layers to prevent that data from ever being something an end user could see, use, alter etc.